screwdriver/api/api.go

```diff
@@ -33,102 +38,212 @@ type ResponseError struct {
 	default:
-		return nil, fmt.Errorf("Unknown error happen while communicate with Screwdriver API")
+		return nil, fmt.Errorf("Unknown error happen while communicate with Screwdriver API: Statuscode=%d", statusCode)
 	}
 }
 
 // GetCommand returns Command from Screwdriver API
-func (c client) GetCommand(smallSpec *Command) (*Command, error) {
-	cmd := new(Command)
+func (c client) GetCommand(smallSpec *util.CommandSpec) (*util.CommandSpec, error) {
 	uri, err := url.Parse(c.baseURL)
 	if err != nil {
-		return nil, fmt.Errorf("The base Screwdriver API is invalid %q", c.baseURL)
+		return nil, fmt.Errorf("Failed to parse URL on GET: %v", err)
 	}
 	uri.Path = path.Join(uri.Path, "commands", smallSpec.Namespace, smallSpec.Name, smallSpec.Version)
-	req, err := http.NewRequest("GET", uri.String(), strings.NewReader(""))
+
+	// Request to api
+	const contentType = "application/json"
+	// No payload
+	payload := bytes.NewBuffer([]byte(""))
+
+	responseSpec, err := c.httpRequest("GET", uri.String(), contentType, payload)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to create request about command to Screwdriver API: %v", err)
+		return nil, err
 	}
 
-	req.Header.Set("Content-type", "application/json")
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.jwt))
-	res, err := c.client.Do(req)
+	return responseSpec, nil
+}
+
+func specToPayloadBytes(commandSpec *util.CommandSpec) (specBytes []byte, err error) {
+	// Generate yaml payload bytes
+	commandSpecBytes, err := json.Marshal(&commandSpec)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to get command from Screwdriver API: %v", err)
+		return nil, err
 	}
-	defer res.Body.Close()
-	body, err := handleResponse(res)
+	var payload util.PayloadYaml
+	payload.Yaml = string(commandSpecBytes)
+	specBytes, err = json.Marshal(&payload)
+	if err != nil {
+		return nil, err
+	}
+	return
+}
+
+func writeMultipartYaml(writer *multipart.Writer, specPath string, commandSpec *util.CommandSpec) error {
+	// Convert command spec to payload
+	specBytes, err := specToPayloadBytes(commandSpec)
+	if err != nil {
+		return fmt.Errorf("Failed to convert spec to bytes:%v", err)
+	}
+
+	// Set yaml filename
+	fileNameYaml := filepath.Base(specPath)
+	yamlPart, err := writer.CreateFormFile("file", fileNameYaml)
+	if err != nil {
+		return fmt.Errorf("Failed to load spec file:%v", err)
+	}
+	// Write yaml part
+	yamlPart.Write(specBytes)
+
+	return nil
+}
+
+func writeMultipartBin(writer *multipart.Writer, commandSpec *util.CommandSpec) error {
+	// Load binary from file
+	fileContentsBin, err := util.LoadByte(commandSpec.Binary.File)
+	if err != nil {
+		return fmt.Errorf("Failed to load binary file:%v", err)
+	}
+
+	// Set filename of binary
+	fileNameBin := filepath.Base(commandSpec.Binary.File)
+	binPart, err := writer.CreateFormFile("file", fileNameBin)
+	if err != nil {
+		return fmt.Errorf("Failed to create form of binary:%v", err)
+	}
+
+	// Write binary part
+	binPart.Write(fileContentsBin)
```

Comment:
I think it will be better to check error.

Reply:
Thank you! I've fixed.